### ISSUE TITLE
Allow editing the default directory for pull-request worktrees from the home folder to a specified folder

### ISSUE THREAD

**Issue by JSpenced** [magit/forge#463]
Maybe, this is possible, but I couldn't find it by looking through all the forge/magit documentation and I checked the variables you can set for each in emacs as well.

When I click `n` to create worktree from pull request:
![image](https://user-images.githubusercontent.com/43424787/153308873-4e097332-6df2-48cc-bdcc-0bec0c12ccab.png)

Then, I select a pull request. I always get the following where it defaults to the home directory:
![image](https://user-images.githubusercontent.com/43424787/153308940-4b34ef92-7ff7-4c0d-bc45-e0e5ac72c751.png)

Normally, I place my worktrees in `~/worktrees` so I am wondering if this can be made a setting for the default directory for pull requests or is this possible already and I missed it?

**Comment by tarsius** [comment]
(Your second screenshot is odd. Without additional context it look like the test that would usually be displayed in the minibuffer is instead dislayed in the mode-line and the editable part is again displayed in the minibuffer. If you do indeed use a package that does something like that, then I would recommend using screenshots with more context. Or disable that package before taking the screenshot. Or don't use screenshots.)

**Comment by tarsius** [comment]
> I always get the following where it defaults to the home directory ~/{user}-{PR#}/branch:

You get something like this, where `|` is the cursor. `~/PATH/|NUMBER-BRANCH`. `PATH` is the directory that contain the repository, e.g. for `~/git/tools/foo` you would get `~/git/tools`. `NUMBER` is the pull-request number and `BRANCH` is the name the contributor chose for their branch.

The latter may contain `/`. If so, then that should probably be replace with `-`, but currently that isn't being done.

I am considering adding a variable to allow users to use their own function to calculate the initial minibuffer input.

**Comment by tarsius** [comment]
You can now use your own function to read the worktree directory.

**Comment by JSpenced** [comment]
Yeah, I was confused myself, but you were correct that it was `{PR#}-BRANCH` just the branch contained `/`'s. Swiper/IDO forced the input to look weird in the screenshots. 

The primary issue I had is that it wasn't going to the repository home `~/git/tools` from your example above. It would always default to the Home directory (`~`) for pull requests. This wasn't true if I did a normal branch checkout in the magit worktree menu (this defaults to the repo directory).

Thanks for the addition! I will make my own function.
